testing/scripts/deploy_staging.sh

```
@@ -0,0 +1,250 @@
+    fi
+
+    DB_DSN="${STAGING_DB_DSN:-}"
+    DB_USER="${STAGING_DB_USER:-}"
+    DB_PASS="${STAGING_DB_PASS:-}"
+
+    if [[ -z "$DB_DSN" || -z "$DB_USER" || -z "$DB_PASS" ]]; then
+        warn "STAGING_DB_DSN, STAGING_DB_USER, or STAGING_DB_PASS not set"
+        warn "Tokens left as placeholders in config.php — edit manually on the server"
+    else
+        ssh "$SSH_HOST" "sed -i \
+            -e 's|__DB_DSN__|${DB_DSN}|g' \
```

Shellcheck (0.11.0)

[info] Note that, unescaped, this expands on the client side.

(SC2029)